snapcraft/internal/indicators.py

```diff
@@ -30,7 +30,11 @@ def download_requests_stream(request_stream, destination, message=None):
     if not message:
         message = 'Downloading {!r}'.format(os.path.basename(destination))
 
-    total_length = int(request_stream.headers.get('Content-Length', '0'))
+    if (request_stream.headers.get('Content-Encoding') == 'gzip'):
```

Comment:
Why not make this the behavior for any `Content-Encoding`?

Such as:
```python
if request_stream.headers.get('Content-Encoding', ''):
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding